ecosystem/typescript/sdk/src/plugins/fungible_asset_client.ts

```diff
@@ -0,0 +1,105 @@
+import { AptosAccount } from "../account";
+import { RawTransaction } from "../aptos_types";
+import * as Gen from "../generated/index";
+import { OptionalTransactionArgs, Provider } from "../providers";
+import { TransactionBuilderRemoteABI } from "../transaction_builder";
+import { MaybeHexString, HexString } from "../utils";
+
+export class FungibleAssetClient {
+  provider: Provider;
+
+  readonly assetType: string = "0x1::fungible_asset::Metadata";
+
+  /**
+   * Creates new FungibleAssetClient instance
+   *
+   * @param provider Provider instance
+   */
+  constructor(provider: Provider) {
+    this.provider = provider;
+  }
+
+  /**
+   *  Transfer `amount` of fungible asset from sender's primary store to recipient's primary store.
+   *
+   * Use this method to transfer any fungible asset including fungible token.
+   *
+   * @param sender The sender account
+   * @param fungibleAssetMetadataAddress The fungible asset address.
+   * For example if you’re transferring USDT this would be the USDT address
+   * @param recipient Recipient address
+   * @param amount Number of assets to transfer
+   * @returns The hash of the transaction submitted to the API
+   */
+  async transferFromPrimaryFungibleStore(
+    sender: AptosAccount,
+    fungibleAssetMetadataAddress: MaybeHexString,
+    recipient: MaybeHexString,
+    amount: number | bigint,
+    extraArgs?: OptionalTransactionArgs,
+  ): Promise<string> {
+    const rawTransaction = await this.generateTransferFromPrimaryFungibleStore(
+      sender,
+      fungibleAssetMetadataAddress,
+      recipient,
+      amount,
+      extraArgs,
+    );
+    const txnHash = await this.provider.signAndSubmitTransaction(sender, rawTransaction);
+    return txnHash;
+  }
+
+  /**
+   * Get the balance of a fungible asset from the account's primary fungible store.
+   *
+   * @param account Account that you want to get the balance of.
+   * @param fungibleAssetMetadataAddress The fungible asset address you want to check the balance of
+   * @returns Promise that resolves to the balance
+   */
+  async getBalanceFromPrimaryFungibleStore(
```

Comment:
I'm not really a fan of appending FromPrimaryFungibleStore to every single function here. Although we want to distinguish from functions interacting directly with store addresses, I think a better way to name and distinguish these are to name them as:
1. Without any suffix, such as getBalance for anything interacting with the (default) primary store
2. Add a suffix [...]ForStore such as getBalanceForStore (or alternatively a Store suffix such as getStoreBalance) for functions interacting with a store address specifically

Reply:
getPrimaryBalance may be a simpler naming for this one

Reply:
That works too